From a7ef557cb3644b4d19030d90ef8b772defdc23dc Mon Sep 17 00:00:00 2001
From: Liming Wang <liming.wang@windriver.com>
Date: Tue, 25 Mar 2014 15:16:35 +0800
Subject: [PATCH 094/628] watchdog: xilinxps: __devinit, __devexit,
 __definitdata, __devexit_p, __exit_p

git://github.com/Xilinx/linux-xlnx.git xilinx-v14.7
commit 068160c0 Merge commit 'v3.8' into master-next

cherry picked the wdt driver modification.

Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 drivers/watchdog/xilinx_wdtps.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/watchdog/xilinx_wdtps.c b/drivers/watchdog/xilinx_wdtps.c
index d4d431a..525082f 100644
--- a/drivers/watchdog/xilinx_wdtps.c
+++ b/drivers/watchdog/xilinx_wdtps.c
@@ -273,7 +273,7 @@ static struct notifier_block xwdtps_notifier = {
  *
  * It does all the memory allocation and registration for the device.
  */
-static int __devinit xwdtps_probe(struct platform_device *pdev)
+static int xwdtps_probe(struct platform_device *pdev)
 {
 	struct resource *regs;
 	int res;
@@ -492,7 +492,7 @@ static int xwdtps_resume(struct device *dev)
 
 static SIMPLE_DEV_PM_OPS(xwdtps_pm_ops, xwdtps_suspend, xwdtps_resume);
 
-static struct of_device_id xwdtps_of_match[] __devinitdata = {
+static struct of_device_id xwdtps_of_match[] = {
 	{ .compatible = "xlnx,ps7-wdt-1.00.a", },
 	{ /* end of table */}
 };
@@ -501,7 +501,7 @@ MODULE_DEVICE_TABLE(of, xwdtps_of_match);
 /* Driver Structure */
 static struct platform_driver xwdtps_driver = {
 	.probe		= xwdtps_probe,
-	.remove		= __exit_p(xwdtps_remove),
+	.remove		= xwdtps_remove,
 	.shutdown	= xwdtps_shutdown,
 	.driver		= {
 		.name	= "xwdtps",
-- 
1.7.5.4

